From fd8137f281f83312c1964f70166286d6fc255823 Mon Sep 17 00:00:00 2001
From: Harish Rachakonda <quic_rachakon@quicinc.com>
Date: Mon, 19 Aug 2024 09:29:38 +0530
Subject: [PATCH] hostapd:Enabling ht_enabled when NL attribute is received

Currently during channel switch ht_enabled is enabled for 6G because of
which hostapd_cli -i wlanX status is showing wrong values

Fix this issue by enabling ht_enabled when NL80211_ATTR_WIPHY_CHANNEL_TYPE
is received.

Patch-dependency: None
Patch-rework: None

Signed-off-by: Harish Rachakonda <quic_rachakon@quicinc.com>
---
 src/drivers/driver_nl80211_event.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/src/drivers/driver_nl80211_event.c b/src/drivers/driver_nl80211_event.c
index 6edaa6c..3a7dee1 100644
--- a/src/drivers/driver_nl80211_event.c
+++ b/src/drivers/driver_nl80211_event.c
@@ -1212,7 +1212,7 @@ static void mlme_event_ch_switch(struct i802_bss *bss,
 {
 	struct wpa_driver_nl80211_data *drv = bss->drv;
 	union wpa_event_data data;
-	int ht_enabled = 1;
+	int ht_enabled = 0;
 	int chan_offset = 0;
 	int link_id = -1;
 	struct i802_link *mld_link = bss->flink;
@@ -1242,6 +1242,7 @@ static void mlme_event_ch_switch(struct i802_bss *bss,
 
 	if (type) {
 		enum nl80211_channel_type ch_type = nla_get_u32(type);
+		ht_enabled = 1;
 
 		wpa_printf(MSG_DEBUG, "nl80211: Channel type: %d", ch_type);
 		switch (ch_type) {
-- 
2.34.1

